### PR TITLE
feat: NFT buy flow

### PR DESCRIPTION
**Pull request by Chef-Cheems**

### CODE REVIEW

File: src/config/constants/tokens.ts [hunk 1]
```diff
@@ -1639,7 +1639,7 @@ export const mainnetTokens = {
 export const testnetTokens = {
   wbnb: new Token(
     TESTNET,
-    '0xae13d989daC2f0dEbFf460aC112a837C89BAa7cd',
+    '0x094616F0BdFB0b526bD735Bf66Eca0Ad254ca81F',
```
Comment (Chef-Cheems) [inline src/config/constants/tokens.ts:1642]: Testnet WBNB that is used in testnet NFT market contract

File: src/state/nftMarket/helpers.ts [hunk 2]
```diff
@@ -84,7 +84,7 @@ export const getNftsFromCollectionSg = async (collectionAddress: string): Promis
       `,
       { collectionAddress: collectionAddress.toLowerCase() },
     )
-    return res.collection.NFTs
+    return res.collection.nfts
```
Comment (Chef-Cheems) [inline src/state/nftMarket/helpers.ts:87]: subgraph changed casing for this key

File: src/views/Nft/market/components/BuySellModals/SellModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import React from 'react'
+
+const BuyModal: React.FC = () => {
```
Comment (Chef-Cheems) [inline src/views/Nft/market/components/BuySellModals/SellModal/index.tsx:3]: Just a placeholder for sell modal. I pre-defined the code structure in this PR.

File: src/views/Nft/market/components/BuySellModals/shared/ApproveAndConfirmStage.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+interface ApproveAndConfirmStageProps {
+  isApproved: boolean
+  isApproving: boolean
+  isConfirming: boolean
+  handleApprove: () => void
+  handleConfirm: () => void
+}
+
+// Shown if user wants to pay with WBNB and contract isn't approved yet
+const ApproveAndConfirmStage: React.FC<ApproveAndConfirmStageProps> = ({
```
Comment (Chef-Cheems) [inline src/views/Nft/market/components/BuySellModals/shared/ApproveAndConfirmStage.tsx:15]: This component will be shared with sell modal

File: src/views/Nft/market/components/BuySellModals/shared/ConfirmStage.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import { useTranslation } from 'contexts/Localization'
+
+interface ConfirmStageProps {
+  isConfirming: boolean
+  handleConfirm: () => void
+}
+
+// Shown in case user wants to pay with BNB
+// or if user wants to pay with WBNB and it is already approved
+const ConfirmStage: React.FC<ConfirmStageProps> = ({ isConfirming, handleConfirm }) => {
```
Comment (Chef-Cheems) [inline src/views/Nft/market/components/BuySellModals/shared/ConfirmStage.tsx:12]: This component will be shared with sell modal

File: src/views/Nft/market/components/BuySellModals/shared/TransactionConfirmed.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+import { useTranslation } from 'contexts/Localization'
+import { getBscScanLink } from 'utils'
+import useActiveWeb3React from 'hooks/useActiveWeb3React'
+import { Divider } from './styles'
+
+interface TransactionConfirmedProps {
+  txHash: string
+}
+
+const TransactionConfirmed: React.FC<TransactionConfirmedProps> = ({ txHash }) => {
```
Comment (Chef-Cheems) [inline src/views/Nft/market/components/BuySellModals/shared/TransactionConfirmed.tsx:12]: This component will be shared with sell modal

File: src/views/Nft/market/components/BuySellModals/shared/styles.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import styled from 'styled-components'
+import { Flex, Image } from '@pancakeswap/uikit'
+
+export const RoundedImage = styled(Image)`
```
Comment (Chef-Cheems) [inline src/views/Nft/market/components/BuySellModals/shared/styles.tsx:4]: Here live the components that will be also used in Sell modal. Components specific to modals live in their directory's styles.tsx

File: src/views/Nft/market/BuyModalDemo/index.tsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  )
+}
+
+const BuyModalDemo = () => {
+  const nfts = useNftsFromCollection('0x60935f36e4631f73f0f407e68642144e07ac7f5e')
+  if (!nfts) return null
+
+  const nftList = Object.values(nfts).filter((nft) => {
+    const tokens = Object.values(nft.tokens)
+    const isTradable = tokens.find((t) => t?.isTradable)
```
Comment (hachiojidev) [inline src/views/Nft/market/BuyModalDemo/index.tsx:30]: Not a blocker but try and void these one-letter variable names.

File: src/views/Nft/market/components/BuySellModals/BuyModal/ReviewStage.tsx [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+          <Message p="8px" variant="danger">
+            <Text>
+              {t('Not enough %symbol% to purchase this NFT', {
+                symbol: paymentCurrency === PaymentCurrency.BNB ? 'BNB' : 'WBNB',
+              })}
+            </Text>
+          </Message>
+        )}
+        <Flex alignItems="center">
+          <Text my="16px" mr="4px">
```
Comment (hachiojidev) [inline src/views/Nft/market/components/BuySellModals/BuyModal/ReviewStage.tsx:80]: Try putting spacing on the parent container.

**Review by hachiojidev — APPROVED** [review]

**Pull request merged**